### ISSUE TITLE
Improved GitHub Integration

### ISSUE THREAD

**Issue by endocytosis** [stackblitz/bolt.new#2367]
A number of users have requested improved GitHub integration for a more fluid development workflow. Currently, Bolt does not support private GitHub repositories. The team is hard at work continuing to improve this process!

You can open a Bolt project in StackBlitz (click Open in StackBlitz at the top-right of the page) and push to git from there. We will track feature requests for tighter GitHub integration here.

**Comment by codewithtyler** [comment]
As noted in [stackblitz/core #3121](https://github.com/stackblitz/core/issues/3121) the StackBlitz GitHub integration has issues creating the GitHub repo. It would be great if somewhere in the Bolt app there was a button to Sync with GitHub. I could imagine and if you clicked the button and the project wasn't connected to a GitHub repo it would prompt you to create one but if it was then it would just push the code up to GitHub automatically. Having a feature like that directly in bolt would be very useful for those users with limited to no development experience.

**Comment by endocytosis** [comment]
Hi @codewithtyler, 

Appreciate the feedback and agreed that continued tighter integration would be beneficial! Given the significant number of performance updates and feature requests engineering is hard at work on production is being prioritized at this time. Will update here as more info become available.

**Comment by codewithtyler** [comment]
@endocytosis any objection to me looking into this for the open source repo? Maybe if I can get it working for the open source version y'all (sorry I'm southern lol) can take that and pull it into the production version.

**Comment by oneshot-engineering** [comment]
say i started a brand new project in bolt, opened in stackblitz, created repository in github and my team made some changes in github and i pulled new changes from github inside Stackblitz.

Now when i open the updated project in bolt from stackblitz -> continue to make changes in bolt -> open the project in stackblitz -> I won't be able to able to push the new changes to the same repository.  

Apparently stackblitz is asking me to create new github repository. 

Any thoughts on how to streamline the workflow of pushing the bolt updates to the same github repository?

**Comment by codewithtyler** [comment]
At the moment there's not really an ideal solution to solve this problem. That is why this issue exists so that the team has something which to gather feedback and come up with a proposed solution.

**Comment by rich-micro** [comment]
Ummm can't you just ask Bolt.new to upgrade the Bolt.new app to include Github integration??? Seriously, why won't you just create the singularity already??? You're like 99% the way there...

Oh no wait, you made it on Github!! Bugger

**Comment by rich-micro** [comment]
Also, +1 for Gitlab integration too please for us professionals (or just git in general)

**Comment by jcmoore** [comment]
I want this badly.  Actually, I kinda expected it to work already!

- From a bolt chat I went to "open in stackblitz"
- From stackblitz I saw connect with github button
- After installing github app I was prompted to upgrade to a team tier account
- 💸

I guess I get a free t-shirt as my consolation prize (...and several million tokens of course)

**Comment by dmenchaca** [comment]
I love building with Bolt but without git using Bolt for production code is a public hazard.

In an ideal world, I would love to be able to commit directly from Bolt, create branches and easily rollback my code directly from Github.

Lovable and Replit already tackled Git on their AI code generators, please team 🙏

**Comment by tejas3732** [comment]
> I love building with Bolt but without git using Bolt for production code is a public hazard.
> 
> In an ideal world, I would love to be able to commit directly from Bolt, create branches and easily rollback my code directly from Github.
> 
> Lovable and Replit have already tackled Git sync on their AI code generators, please team prioritize this🙏

Absolutely

**Comment by electricmonk** [comment]
+100

**Comment by jcmoore** [comment]
tl;dr: maybe try https://www.npmjs.com/package/boltops

Hey y'all, forgot to share that I set up a very rough -- and poorly documented -- `npx` package to copy files to bolt webcontainers from my host machine and back again (link above).

Using it, I leverage git locally, and have a reasonable flow with git diff/add/reset to see what changes bolt made without having to do a full commit-push-pull cycle (muddying up git history).

Pull requests (esp. to improve docs) are most welcome.

**Comment by carloslfu** [comment]
This is so cool @jcmoore! Syncing from the inside is a clever trick! I'll give it a try!

**Comment by bizrockman** [comment]
@jcmoore At least a workaround. 

I would even go with only having Stackblitz in the background, but you can not really sync files there as well. Since Stackblitz already supported Github integration, it would be great to link that workspace to bolt.new, so that I can make github operations there.

Right now I can only import a Github or reuse the code I was working in the chat. But both resolving in that I need to create a new GitHub Repo instead using the existing ones.

So thanks again for this workaround, that I can at least use a local dev env to sync files and than git hub

**Comment by ForestPenland** [comment]
I would love to see something similar to what replit does, in the bolt editor I would love a fully functional console.... Edit code in my own IDE, pull changes back to bolt and vise versa.

**Comment by davecazz** [comment]
adding my voice. we just purchased a teams license for stackblitz hoping we would be able to use bolt on our private repos. huge bummer that it's not implemented or clear guidance that it's on the engineering roadmap other than, "we are gathering feedback". even just the idea of having stackblitz be the intermediator between github and bolt would be better than nothing.

**Comment by Nathanglobalsense** [comment]
Also, +1 for Gitlab integration too please ! (or just git in general)

**Comment by codewithtyler** [comment]
@davecazz it is on the roadmap. I know there's been at least 2 meetings so far in the past couple of weeks to discuss Git integration. The hope is that we'll get something in Bolt this month but I can't make any promises on that as I don't know how far long they are on it.

@Nathanglobalsense it's my understanding that general git has been suggested to the team rather than focusing on GitHub specifically. So we'll just have to wait and see what gets implemented.

**Comment by kivio** [comment]
I support adding this feature. Currently, I face a challenge managing code across three different platforms:

Bolt.new project
StackBlitz project
GitHub repository

Each platform maintains its own separate state of the code, which creates inconsistency. I propose integrating GitHub as the single source of truth across all three tools:

GitHub should serve as the main codebase repository
StackBlitz, VS Code, and Bolt should all have direct Git command access to this repository
An automatic commit feature would be beneficial, especially for AI-generated changes:

Each AI modification would trigger an automatic commit
The commit message would describe the AI's changes
This would create a traceable history of modifications
It would make it easier to recover code when AI changes cause issues



This integration would help prevent code inconsistencies and make it easier to track and reverse unwanted changes.

**Comment by bentokyan77** [comment]
I support this thread and I'm missing this feature too! 

From bolt.new I go to stackblitz, but instead of connecting to a specific repository, I can only create a new one. Which doesn't make any sense at all. 

Please give this as much attention and priority as possible.

**Comment by carloslfu** [comment]
Many of my most serious clients are migrating to Lovable because of this missing feature.

**Comment by lucasgates** [comment]
This feature for me will make or break bolt.new for me. If I were to copy and paste from bolt to stackblitz, I might as well use ChatGPT directly. I really love bolt.new, and hope you all can get this done as quickly as possible.

**Comment by gravity-matt** [comment]
+1

**Comment by jefersonbrain** [comment]
+1

**Comment by googelybear** [comment]
+1

**Comment by anhtuan9622** [comment]
+1

**Comment by THALLIVA** [comment]
+1

**Comment by navid-a** [comment]
This remains to be the biggest pain point for me, to the point that I am considering switching although I have my entire project here.

I would like:
+ Bolt to push all updates to github so I can start modifying the code on my preferred IDE locally.
+ To be able to change the code on my side and push to github and for bolt to be able to see the updates.

Another strongly needed optional requirement:
+ Be able to import a whole new github project into bolt and start working with bolt from that.

I hope you can prioritize this in your upcoming work. This is also crucial for teams to be able to work on a project together.

**Comment by NitsanCohen770** [comment]
I really don't get it why this feature is not there already 
this will easily add millions of developers to use bolt

**Comment by Hiaslafson** [comment]
Any updates on this? thinking of switching to lovable... Really would need to checkout my stuff via github...

**Comment by slafaurie97** [comment]
Same here!! Want to use Bolt but this github integraiton is crucial

**Comment by dahelg** [comment]
I have come across this problem and am really wondering why this is not already implemented. I really thought I was doing something wrong. But it is indeed a real problem and should be solved as it makes no sense the way it is currently solved.

**Comment by epodak** [comment]
Where is the "Open in Bolt.new" button?

**Comment by kayuzee** [comment]
Saw this random chrome extension when researching this topic on Reddit that might help some
https://chromewebstore.google.com/detail/bolt-to-github/pikdepbilbnnpgdkdaaoeekgflljmame?hl=en

**Comment by nirkons** [comment]
stackblitz to git not working properly because once you go back to bolt and click open in stackblitz it loses the connection to the git project and there is no way to save it

**Comment by tejas3732** [comment]
I have got the beta invitation to Github Sync, and it's awesome. Works pretty well. The Github sync feature is coming very soon within 1 week, as the founder told me. So hold your horses :)

**Comment by codewithtyler** [comment]
This has officially been released and is now available for everyone to use.
